File: ch07/VbServerControl1/Listing07-37.vb
Imports System.ComponentModel
Imports System.Web.UI
Imports System.Web.UI.WebControls

<DefaultProperty("Text"),
ToolboxData("<{0}:ServerControl39 runat=server></{0}:ServerControl39>")>
Public Class Listing0737
    Inherits WebControl

    <Bindable(True)>
    <Category("Appearance")>
    <DefaultValue("")>
    <TypeConverter(GetType(GuidConverter))>
    Property BookId() As System.Guid

    Protected Overrides Sub RenderContents(ByVal output As HtmlTextWriter)
        output.Write(BookId.ToString())
    End Sub

End Class

File: ch07/CSharpServerControl1/Listing07-37.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace CSharpServerControl1
{
    [DefaultProperty("Text")]
    [ToolboxData("<{0}:Listing0737 runat=server></{0}:Listing0737>")]
    public class Listing0737 : WebControl
    {
        [Bindable(true)]
        [Category("Appearance")]
        [DefaultValue("")]
        [TypeConverter(typeof(GuidConverter))]
        public Guid BookId { get; set; }

        protected override void RenderContents(HtmlTextWriter output)
        {
            output.Write(BookId.ToString());
        }
    }
}